src/include/common/data_chunk/data_chunk.h

```diff
@@ -33,7 +33,7 @@ class DataChunk {
 
     inline uint32_t getNumValueVectors() const { return valueVectors.size(); }
 
-    inline std::shared_ptr<ValueVector> getValueVector(uint64_t valueVectorPos) {
+    inline const std::shared_ptr<ValueVector>& getValueVector(uint64_t valueVectorPos) const {
```

Comment:
Changed this function to return a reference to the `std::shared_ptr` to avoid triggering the reference counting logic, I looked through the uses of this function and I don't think any of them do anything with the `std::shared_ptr` object. When parsing from a single column with `IGNORE_ERRORS=true` (which calls `getValueVector()` 6 times per value which is the worst case) this speeds up the overall copy by 5-10%.

Reply:
I'll do this in a separate PR since this will involve some more heavy refactoring https://github.com/kuzudb/kuzu/issues/4313